src/fifo_map.hpp

```diff
@@ -86,11 +92,12 @@ class fifo_map_compare
         keys->erase(key);
     }
 
+    /// the next valid insertion timestamp
+    size_t timestamp = 1;
```

Comment:
Is there a reason to make the timestamp public?

Reply:
No particular reason - I'll add a getter for current value.

Reply:
Is that useful?

Reply:
Yes, it's used below to transfer current timestamp value to new instance during copying.

Reply:
I understand. But I do not see the need to make the variable public in the first place. Why would client code need access to the variable?

Reply:
Would you suggest a `friend` declaration in this case to make it completely hidden from public access?

Reply:
In the code, I see you only call the getter in the copy constructor. That copy constructor can work directly with the private member. I think we can make `timestamp` private and remove the getter. What class should required a `friend` declaration?

Reply:
Perhaps I am missing something and will appreciate some guidance. Sorry for making this discussion for such a simple patch longer than expected.
 
My observation is that scope access prevents what you propose, as `m_timestamp` is a private member of `nlohmann::fifo_map_compare`. It's a compile time error, since private variables of `nlohmann::fifo_map_compare` are not accessible from `nlohmann::fifo_map`. 

I guess that this can compile in your unit tests, as there's a good old `#define private public` in `tests/unit.cpp`.

Reply:
You are absolutely right. Then `nlohmann::fifo_map_compare` should be a friend class. (It's been a while that I wrote the code...)

Reply:
Fixed in `c0ea4f4` below.